### PR TITLE
Add Enhanced Model

### LINKED ISSUE #282: 💡[FEATURE]: Model Enhancement By performing Hyperparameter tuning
**Is your feature request related to a problem? Please describe.**
The current credit risk prediction is trained over low accuracy , I would be performing hyperparameter tuning to enhance the model's prediction over the data set.

**Describe the solution you'd like**
This task requires analysis and understanding of the dataset that has been used and restucture entire model.

### PR DESCRIPTION
**Pull request by binguliki**
## 🛠️ Related Issue

- Closes #282 

## 📝 Description

The existing neural network model for `CREDIT RISK PREDICTION` wasn't trained properly , so first i replaced it with a complex model and tried to overfit the model over the train data. Once over fitted i tried to regularize it using different regularization techniques like L1 , L2 , dropouts etc and performed Hyperparameter tuning , I have also used Keras_tuner to find the best hyperparameters . 

## 🔍 Type of PR
- [X] ✨ Feature enhancement

## ✅ Checklist

- [X] I have performed a self-review of my code.
- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X] I have commented my code, particularly in hard-to-understand areas.

## ℹ️ Additional Context

To improve the performance and get human level performance an extensive data analysis can be performed to get better outcomes as the model has achieved maximum performance over the current state of the data

### CODE REVIEW

**Comment by binguliki** [comment]
@Akshat111111 , Hey Bro can you please review the Code , I have performed Proper optimization of the neural network. Please provide me a `level 3` under `GSSoC'24`.

**Review by Akshat111111 — COMMENTED** [review]
yeah, it works
